core/field_variable.js

```diff
@@ -41,13 +41,16 @@ goog.require('goog.string');
  *     a unique variable name will be generated.
  * @param {Function=} opt_validator A function that is executed when a new
  *     option is selected.  Its sole argument is the new option value.
+ * @param {!Array.<string>} variableTypes A list of the types of variables to
```

Comment:
I have it set to [''] in getVariableTypes_ where it checks if this.variableTypes is null. If variableTypes is null and I do
  "this.variableTypes = opt_variableTypes || [''"];
in this Block.FieldVariable function it will override the "null" value with [""].
By waiting until getVariableTypes_ to set the default to [""] we can preserve the null.